src/lang/errors.ts

```diff
@@ -336,16 +359,44 @@ export function kclErrorsToDiagnostics(
   let nonFatal: CodeMirrorDiagnostic[] = []
   const errs = errors
     ?.filter((err) => isTopLevelModule(err.sourceRange))
-    .map((err): CodeMirrorDiagnostic => {
+    .flatMap((err) => {
+      const diagnostics: CodeMirrorDiagnostic[] = []
+      let message = err.msg
+      if (err.kclBacktrace.length > 0) {
+        // Show the backtrace in the error message.
+        for (let i = 0; i < err.kclBacktrace.length; i++) {
+          const item = err.kclBacktrace[i]
+          if (
+            i > 0 &&
+            isTopLevelModule(item.sourceRange) &&
+            item.sourceRange[0] !== err.sourceRange[0] &&
+            item.sourceRange[1] !== err.sourceRange[1]
+          ) {
+            diagnostics.push({
+              from: item.sourceRange[0],
+              to: item.sourceRange[1],
+              message: 'Part of the error backtrace',
+              severity: 'hint',
+            })
```

Comment:
Is it possible to have two back traced errors?
Probably not hey?
Just wondering if it's possible for this to be confusing because which back trace is it referring to.

Other thought,  we could put the index somewhere in the message, but come to think of it, probably good without it, confusing otherwise, and you should go to the main diagonastic to see the error order.

Reply:
Hints don't get shown for non-fatal errors, so there should only be one backtrace.